qadence2_ir/irast.py

```diff
@@ -291,4 +291,10 @@ def __eq__(self, other: object) -> bool:
         return self._args == other._args and self._attrs == other._attrs
 
     def __repr__(self) -> str:
-        return f"{self._tag}({self._head}, {self._args}, {self._attrs})"
+        result = f"{self._tag}('{self._head}', "
+        for value in self._args:
+            result += f"{value}, "
+        for key, value in self._attrs.items():
+            value_text = f"'{value}'" if isinstance(value, str) else f"{value}"
+            result += f"{key}={value_text}, "
```

Comment:
Any particular reason to use a loop instead of `", ".join(…)` and `repr` to automatically add quotation marks to strings?

```suggestion
       # args
        result += ", ".join(map(str, self._args))
        # attributes
        result += ", ".join([f"{key}={repr(val)}" for key, val in self._attrs.items()])
```